lib/src/Pos/Launcher/Configuration.hs

```diff
@@ -107,6 +109,28 @@ instance FromJSON Configuration where
 
             -- else default to RequiresMagic
             | otherwise -> pure RequiresMagic
+        ccTxValRules <- if
+            | HM.member "txValidationRules" o -> do
+                valRulesObj <- o .: "txValidationRules"
+
+                attribResEpoch <- valRulesObj .: "attribResrictEpoch"
+                aaSize <- valRulesObj .: "addrAttribSize"
+                taSize <- valRulesObj .: "txAttribSize"
+                -- attribResEpoch is used twice because
+                -- the second field of TxValidationRules
+                -- is for the current Epoch, which is taken
+                -- from the state later on.
+                pure $ TxValidationRules
+                           (EpochOrSlot . Left $ EpochIndex attribResEpoch)
+                           (EpochOrSlot . Left $ EpochIndex attribResEpoch)
+                           aaSize
+                           taSize
+            -- Defaults to this value in order for golden tests to pass
+            | otherwise -> pure $ TxValidationRules
+                                      (EpochOrSlot . Left $ EpochIndex 0)
+                                      (EpochOrSlot . Left $ EpochIndex 0)
+                                      1000
+                                      1000
```

Comment:
Having an implicit default seems potentially dangerous, but in this case would it be immediately noticeable that something was amiss upon system start? (If a production or testnet configuration was missing the section, and this default was used)

Reply:
I agree.

Reply:
I could put a check in `checkTx` to check for a specific attribute size, something like 9999999999, and make it fail on that with a relevant error message.

Reply:
Added comments in another review here: https://github.com/input-output-hk/cardano-sl/pull/3878#pullrequestreview-182949475